### PR TITLE
[kots]: add preflight checks for IDE/meta nodes

### LINKED ISSUE #11347: Create KOTS preflight checks

### PR DESCRIPTION
**Pull request by MrSimonEmms**
## Description

Add preflight checks to the IDE/meta nodes as per the requirements

## Related Issue(s)

Fixes #11347

## How to test

Deploy via KOTS

## Release Notes

```release-note
[kots]: add preflight checks for IDE/meta nodes
```

## Werft options:

- [ ] /werft with-preview

### CODE REVIEW

File: install/kots/manifests/kots-preflight.yaml [hunk 1]
```diff
@@ -237,6 +261,30 @@ spec:
...
               message: There are at least 4 cores in the nodes
+    - nodeResources:
+        checkName: Memory per workload_meta node
+        filters:
+          selector:
+            matchLabel:
+              gitpod.io/workload_meta: "true"
+        outcomes:
+          - fail:
+              when: "min(memoryCapacity) < 16G"
```
Comment (Pothulapati) [inline install/kots/manifests/kots-preflight.yaml:272]: Isn't this a bit too high? 🤔 

From the https://github.com/gitpod-io/gitpod/issues/11344#issuecomment-1185493784, It does make sense but should we instead warn users on this? Fialing seems to be a bit hard requirement🤔

Comment (geropl) [reply install/kots/manifests/kots-preflight.yaml:272]: @MrSimonEmms If think `< 4c/16GB` should be a warning. hard limit on `2c/8GB` make sense.

**Review by Pothulapati — APPROVED** [review]
LGTM :shipit:

**Pull request merged**